lib/openstack_query/query_factory.py

```diff
@@ -0,0 +1,37 @@
+from typing import Type
+from openstack_query.mappings.mapping_interface import MappingInterface
+from openstack_query.query_blocks.query_builder import QueryBuilder
+from openstack_query.query_blocks.query_output import QueryOutput
+from openstack_query.query_blocks.query_parser import QueryParser
+from openstack_query.query_blocks.query_executer import QueryExecuter
+from structs.query.query_components import QueryComponents
+
+
+# pylint:disable=too-few-public-methods
+
+
+class QueryFactory:
+    """
+    This class is used to construct a query object using a factory pattern.
+    Decouples the API from the implementation
+    """
+
+    @staticmethod
+    def build_query_deps(mapping_cls: Type[MappingInterface]) -> QueryComponents:
```

Comment:
We can probably just move this to a module method, rather than having a class for it

I think it's a hang-over from when I was trying to do it with metaclasses

Reply:
I may need this class to do chaining between queries